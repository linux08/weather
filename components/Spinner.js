import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';

const styles = {
    spinnerStyle: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
};

const Spinner = ({ size, color }) => (
    <View style={styles.spinnerStyle}>
        <ActivityIndicator size={size || 'large'} color={color || '#2F3CA3'} />
    </View>
);

Spinner.defaultProps = {
    size: 'large',
    color: '#2F3CA3',
};

Spinner.propTypes = {
    size: PropTypes.string,
    color: PropTypes.string,
};

export default Spinner;