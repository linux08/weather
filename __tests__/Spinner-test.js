import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Spinner from '../components/Spinner';


Enzyme.configure({ adapter: new Adapter() });


function setup() {
    const props = {
        size: 'large',
        color: '#2F3CA3',
    }

    const enzymeWrapper = shallow(<Spinner {...props} />)

    return {
        props,
        enzymeWrapper
    }
}


describe('Spinner snapshot', () => {
    jest.useFakeTimers();
    jest.setTimeout(30000);

    it('should render self and subcomponents', async () => {
        const { enzymeWrapper } = setup()
        expect(enzymeWrapper.prop("style")).toEqual({
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center'
        })
    })

    it('renders the Spinner', async () => {
        const tree = renderer.create(<Spinner />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});

