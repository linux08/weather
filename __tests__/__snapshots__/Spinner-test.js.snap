// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the Spinner 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2F3CA3"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Spinner snapshot renders the Spinner 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2F3CA3"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
